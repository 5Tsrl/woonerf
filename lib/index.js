module.exports = {
  auth0: require('./auth0'),
  fetch: require('./fetch'),
  html: require('./html'),
  mount: require('./mount'),
  store: require('./store')
}
