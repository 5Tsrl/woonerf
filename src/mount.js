import React from 'react'
import {render} from 'react-dom'
import {Provider} from 'react-redux'
import {browserHistory} from 'react-router'
import {syncHistoryWithStore} from 'react-router-redux'

import createStore from './store'

if (process.env.NODE_ENV === 'development') {
  const Perf = window.Perf = require('react-addons-perf')
  Perf.start()
}

export function create ({
  app,
  reducers
}) {
  const store = createStore(reducers)
  const history = syncHistoryWithStore(browserHistory, store)
  return React.createElement(Provider, {store},
    React.createElement(app, {history, store}))
}

export default function mount ({
  app,
  id = 'root',
  reducers
}) {
  return render(
    create({app, reducers}),
    document.getElementById(id)
  )
}
