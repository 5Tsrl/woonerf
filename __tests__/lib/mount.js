/* globals describe, it, expect */

import React, {Component, PropTypes} from 'react'
import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'

import {create} from '../../lib/mount'

class App extends Component {
  static propTypes = {
    history: PropTypes.object.isRequired
  }

  render () {
    return <div />
  }
}

describe('mount', () => {
  it('should mount the react application to root', () => {
    const tree = mount(create({app: App, reducers: {}}))
    expect(mountToJson(tree)).toMatchSnapshot()
  })
})
