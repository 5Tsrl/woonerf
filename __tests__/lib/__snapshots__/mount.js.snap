exports[`mount should mount the react application to root 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }>
  <App
    history={
      Object {
        "createHref": [Function],
        "createKey": [Function],
        "createLocation": [Function],
        "createPath": [Function],
        "getCurrentLocation": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "listenBefore": [Function],
        "push": [Function],
        "replace": [Function],
        "transitionTo": [Function],
        "unsubscribe": [Function],
      }
    }>
    <div />
  </App>
</Provider>
`;
